app/frontend/src/components/Datepicker.tsx

```diff
@@ -42,16 +61,21 @@ export default function Datepicker({
             className={className}
             disableToolbar
             error={error}
-            format="DD.MM.YYYY"
+            format={getLocaleFormat()}
             fullWidth
             helperText={helperText}
             id={id}
+            KeyboardButtonProps={{
+              "aria-label": CHANGE_DATE,
+            }}
             InputLabelProps={{
               shrink: true,
             }}
             label={label}
             minDate={minDate}
-            onChange={(date) => onChange(date?.toDate())}
+            onChange={(date) => {
+              if (date?.isValid()) onChange(date?.toDate());
+            }}
```

Comment:
The previous one is better, because if you leave the field like "05/04/202_" and submit, you get an error, whereas with this change, it just submits the most recent valid date.

However, for some reason the tests don't pass without this change and I couldn't figure out why

Reply:
Does that mean if I do `05/04/2021`, then click around and change it to like `01/06/20__`, it'll still submit `05/04/2021`?

Reply:
If by click around you mean clear it and type "010620" then yes

Reply:
I got curious, revert to the old code locally and when you do that interaction, it crashed the app on this bit of code: `data.birthdate.toISOString().split("T")[0]`, so that might be why the test is failing? I think it's done its job here in catching the problem! :D